aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1156,6 +1156,25 @@ std::vector<Tensor> gradient(const Tensor& self, IntArrayRef dim, int64_t edge_o
 
 // ALL REDUCE #################################################################
 
+inline bool should_use_acc_buffer(at::TensorIterator& iter) {
+  const auto ndim = iter.ndim();
+  if (!iter.device().is_cpu()) {
+    return false;
+  }
+  if (!at::isReducedFloatingType(iter.common_dtype())) {
+    return false;
+  }
+  if (ndim < 2 || iter.noutputs() != 1) {
```

Comment:
it happens with `ndim >= 3` so should check `ndim < 3` here?

Reply:
Thanks for the comment. Fixed.